-code-checks

ERROR: Build errors found during clang-tidy run.
```
SimCalorimetry/HcalSimAlgos/src/HcalAmplifier.cc:47:18: error: use of overloaded operator '==' is ambiguous (with operand types 'DetId' and 'DetId::Detector') [clang-diagnostic-error]
  if (frame.id() == DetId::Hcal && ((frame.id().subdetId() == HcalGenericDetId::HcalGenBarrel) ||
                 ^
/cvmfs/cms-ib.cern.ch/nweek-02640/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-08-03-1100/src/DataFormats/DetId/interface/DetId.h:74:23: note: candidate function
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/SimCalorimetry/HcalSimAlgos/src/HcalAmplifier.cc
+++ b/SimCalorimetry/HcalSimAlgos/src/HcalAmplifier.cc
@@ -44,8 +44,8 @@ void HcalAmplifier::amplify(CaloSamples& frame, CLHEP::HepRandomEngine* engine) 
   }
   pe2fC(frame);
 
-  if (frame.id() == DetId::Hcal && ((frame.id().subdetId() == HcalGenericDetId::HcalGenBarrel) ||
-                                    (frame.id().subdetId() == HcalGenericDetId::HcalGenEndcap))) {
+  if (frame.id().det() == DetId::Hcal && ((frame.id().subdetId() == HcalGenericDetId::HcalGenBarrel) ||
+                                          (frame.id().subdetId() == HcalGenericDetId::HcalGenEndcap))) {
     const HcalSimParameters& params = static_cast<const HcalSimParameters&>(theParameterMap->simParameters(frame.id()));
     if (params.delayQIE() > 0)
       applyQIEdelay(frame, params.delayQIE());

